build.gradle

```diff
@@ -81,10 +111,69 @@ clean {
     return path
 }*/
 
-publishing {
-    publications {
-        mavenJava(MavenPublication) {
-            artifact jar
+// Configure publishing to the Sonatype Maven Central repo
+uploadArchives {
+    repositories {
+        mavenDeployer {
+            beforeDeployment { MavenDeployment deployment -> signing.signPom(deployment) }
+
+            repository(url: "https://oss.sonatype.org/service/local/staging/deploy/maven2/") {
+                authentication(userName: sonatypeUsername, password: sonatypePassword)
+            }
+
+            snapshotRepository(url: "https://oss.sonatype.org/content/repositories/snapshots/") {
+                authentication(userName: sonatypeUsername, password: sonatypePassword)
+            }
+
+            pom.project {
+                name 'Network Survey Messaging'
+                packaging 'jar'
+                // optionally artifactId can be defined here
+                description 'Protobuf and gRPC stubs for sending Wireless Protocol Survey messages.'
+                url 'https://github.com/christianrowlands/network-survey-messaging'
+
+                scm {
+                    url 'https://github.com/christianrowlands/network-survey-messaging'
+                    connection 'scm:git:git://github.com/christianrowlands/network-survey-messaging.git'
+                    developerConnection 'scm:git:ssh://git@github.com:christianrowlands/network-survey-messaging.git'
+                }
+
+                issueManagement {
+                    system 'GitHub'
+                    url 'https://github.com/christianrowlands/network-survey-messaging/issues'
+                }
+
+                licenses {
+                    license {
+                        name 'The Apache License, Version 2.0'
+                        url 'https://github.com/christianrowlands/network-survey-messaging/blob/master/LICENSE'
+                        distribution 'repo'
+                    }
+                }
+
+                developers {
+                    developer {
+                        id 'christianrowlands'
+                        name 'Christian Rowlands'
+                        email 'christian.rowlands@gmail.com'
+                    }
+                }
+            }
         }
     }
+}
+
+/**
+ * Try to locate a credential value (user name or password) in the environment,
+ * or falling back to the credentials store.
+ *
+ * This supports building locally on a developers machine and on a Jenkins server with credentials bound
+ * into the environment.
```

Comment:
Probably should document the pattern for the environment variable name,

Reply:
I am going to just remove it.  It is not necessary at this point.